src/odemis/driver/zeiss.py

```diff
@@ -985,13 +1071,38 @@ def _run_move(self):
                                      1)
                 traveled_dist = total_dist * traveled_ratio
                 self.pos = orig_pos + traveled_dist
+
+                # R is allowed to do any rotation, but always stays within 0 -> 360°
+                self.pos[-2] %= 2 * math.pi
+
             if not self._stage_stop.is_set():
                 self.pos = self.target_pos
+                self.pos[-2] %= 2 * math.pi
+
         except Exception:
             logging.exception("Failed to run the move")
         finally:
             self._is_moving = False
 
+    def _move_to_pos(self, pos):
+        """
+        After setting movement parameters time and position values, start the movement threaded.
+        :param pos: iterable of float of len 3 or 6.
+        """
+        # fill the missing position values by filling the current position
+        pos = numpy.append(pos, self.pos[len(pos):])
+        pos[3:5] = [math.radians(pos[3]), math.radians(pos[4])]
+        self.target_pos = numpy.array(pos)
+        self._start_move = time.time()
+        dist = numpy.sqrt((self.pos - self.target_pos) ** 2)
+        dur = sum(dist) / self._speed
+        self._end_move = time.time() + dur
+        self.dur = dur
+        self._is_moving = True
+        self._stage_stop.clear()
+        self._mover = threading.Thread(target=self._run_move)
```

Comment:
Just a question, how often will `_move_to_pos()` be called during runtime?

Reply:
In some worst cases, about once per second. Typically, it's more like every 10s or less. 
I think you're worried about the cost of creating a new thread every time? I'd say it's fine for this case. It's also only for the simulator. On real systems, this code is not used.

Reply:
Thanks for clearing that up Eric.